log4j-core/src/main/java/org/apache/logging/log4j/core/config/plugins/processor/PluginProcessor.java

```diff
@@ -107,6 +123,93 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
+
+                    if ((methodName.toLowerCase(Locale.ROOT).startsWith("set") // Typical pattern for setters
+                                    || methodName
+                                            .toLowerCase(Locale.ROOT)
+                                            .startsWith("with")) // Typical pattern for setters
+                            && methodElement.getParameters().size()
+                                    == 1 // It is a weird pattern to not have public setter
+                    ) {
+
+                        Types typeUtils = processingEnv.getTypeUtils();
+
+                        boolean followsNamePattern = methodName.equals(
+                                        String.format("set%s", expectedFieldNameInASetter(fieldName)))
+                                || methodName.equals(String.format("with%s", expectedFieldNameInASetter(fieldName)));
+
+                        // Check if method is public
+                        boolean isPublicMethod = methodElement.getModifiers().contains(Modifier.PUBLIC);
+
+                        // Check if the return type of the method element is Assignable.
+                        // Assuming it is a builder class the type of it should be assignable to its parent
+                        boolean checkForAssignable = typeUtils.isAssignable(
+                                methodElement.getReturnType(),
+                                methodElement.getEnclosingElement().asType());
+
+                        boolean foundPublicSetter = followsNamePattern && checkForAssignable && isPublicMethod;
+                        if (foundPublicSetter) {
+                            // Hurray we found a public setter for the field!
+                            return;
+                        }
+                    }
+                }
+            }
+            // If the setter was not found generate a compiler warning.
+            processingEnv
+                    .getMessager()
+                    .printMessage(
+                            Diagnostic.Kind.ERROR,
+                            String.format(
+                                    "The field `%s` does not have a public setter, Note that @SuppressWarnings(\"%s\"), can be used on the field to suppress the compilation error. ",
+                                    fieldName, SUPPRESS_WARNING_PUBLIC_SETTER_STRING),
+                            element);
+        }
+    }
+
+    /**
+     *  Helper method to get the expected Method name in a field.
+     *  For example if the field name is 'isopen', then the expected setter would be 'setOpen' or 'withOpen'
+     *  This method is supposed to return the capitalized 'Open', fieldName which is expected in the setter.
+     * @param fieldName who's setter we are checking.
+     * @return The expected fieldName that will come after withxxxx or setxxxx
+     */
+    public static String expectedFieldNameInASetter(String fieldName) {
```

Comment:
Please make this `private`, we only expose public methods, when we need to.

Reply:
Done